vertexbuffer.cpp

```diff
@@ -0,0 +1,44 @@
+#include "vertexbuffer.h"
+
+#include "renderutils.h"
+
+VertexBuffer::VertexBuffer(const void* data, unsigned int size) :
+    mRendererID(0)
+{
+    GLCall(glGenBuffers(1, &mRendererID));
+    GLCall(glBindBuffer(GL_ARRAY_BUFFER, mRendererID));
+    GLCall(glBufferData(GL_ARRAY_BUFFER, size, data, GL_STATIC_DRAW));
+    mSize = size;
+}
+
+VertexBuffer::VertexBuffer(unsigned int size) :
+    mRendererID(0)
+{
+    GLCall(glGenBuffers(1, &mRendererID));
+    GLCall(glBindBuffer(GL_ARRAY_BUFFER, mRendererID));
+    GLCall(glBufferData(GL_ARRAY_BUFFER, size, nullptr, GL_DYNAMIC_DRAW));
+    mSize = size;
+}
+
+VertexBuffer::~VertexBuffer()
+{
+    GLCall(glDeleteBuffers(1, &mRendererID));
+}
+
+void VertexBuffer::SetBufferData(const void* data) const
```

Comment:
This is interesting question. Should this be const here? It certainly can because it is not affecting the class's internals. This glBufferSubData() is populating the array with data, but on the gpu and not here. However it just seems strange to mark a function called "SetBufferData" as const. Not sure if that leads to some confusing code to read for someone later on.

Reply:
If the internals of the class are updated at all you should not use const.

It appears as if the internals of the class are not updated, so this is fine to use here, but agreed if I were reading it I would kind of question it for a second. Writers choice to leave it on or off